homeassistant/components/pvpc_hourly_pricing/config_flow.py

```diff
@@ -0,0 +1,28 @@
+"""Config flow for pvpc_hourly_pricing."""
+from homeassistant import config_entries
+
+from . import CONF_NAME, UI_CONFIG_SCHEMA
+from .const import ATTR_TARIFF, DOMAIN, UNIQUE_ID_MASK
+
+_DOMAIN_NAME = DOMAIN
+
+
+class TariffSelectorConfigFlow(config_entries.ConfigFlow, domain=_DOMAIN_NAME):
+    """Handle a config flow for `pvpc_hourly_pricing` to select the tariff."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        if user_input is not None:
+            unique_id = UNIQUE_ID_MASK.format(user_input[ATTR_TARIFF])
+            await self.async_set_unique_id(unique_id)
+            self._abort_if_unique_id_configured()
+            return self.async_create_entry(title=user_input[CONF_NAME], data=user_input)
```

Comment:
The tariff needs to be stored in the config entry data.  It cannot be the unique id if you are able to change it later. Consider a higher timeout, maybe 10s.

Reply:
> The tariff needs to be stored in the config entry data. It cannot be the unique id if you are able to change it later. 

Don't follow you there, sorry. The tariff is already in `entry.data` (and in that step, inside the `user_input`)

Reply:
It looks like you fixed this when you removed the options cruft